### DISCUSSION TITLE
How to save image with the whole indexed palette in 14.2.0?

### DISCUSSION THREAD

**Discussion by pryvyd9** [dlemstra/Magick.NET#1763] [Help]
hi, the palette seems to be lost on save regardless of QuantizeSettings. I want to note that I'm saving a file with only 2 colors of the palette as of now but I still need it to save the whole palette.
In 13.6.0 I wrote PAL8 files with a code like this and it saved it correctly without me specifying additional settings
```
cellsMap.Settings.SetDefine("png:color-type", "1");
string[] colormap = [
   "#00FF00",
   "#FF0000",
   "#FFFC00",
  ...
];
cellsMap.Map(colormap.Select(n => new MagickColor(n)));
await cellsMap.WriteAsync(path);
```

In 14.2.0 I specified QuantizeSettings, ColorType and ColorSpace explicitly but it doesn't seem to work
```
          var qs = new QuantizeSettings
          {
                Colors = 256,
                DitherMethod = DitherMethod.No,
            };
            cellsMap.ColorType = ColorType.Palette;
            cellsMap.ColorSpace = ColorSpace.sRGB;
            cellsMap.Remap(colormap.Select(n => new MagickColor(n)), qs);
            await cellsMap.WriteAsync(path, MagickFormat.Png);
```
I also tried to open a template file in correct format and save as Png00 to preserve the format but it's still not preserved. Saving as Png8 still saves as PAL1.

Please, advise how to save Png8 palette in PAL8 format.

**Comment by dlemstra** [comment]
I don't know if I have enough information to give you an example. Can you show a complete example of what used to work and no longer works?

**Reply by pryvyd9** [reply]
here's an example: https://github.com/pryvyd9/PaletteIssue
Run OldVersion which produces result that saves the whole palette
![зображення](https://github.com/user-attachments/assets/df10854d-38b9-413c-ac2a-591e4bb8bd98)
And NewVersion which only saves 2 colors to the palette
![зображення](https://github.com/user-attachments/assets/77858f8c-c3cc-42c7-9a0a-d76ede4fb2a3)

I want to force new version to save all colors (ColormapSize = 18). Is there a way?
UPD: updated issue title to better reflect the question

**Reply by dlemstra** [reply]
I finally had some time to take a look at this. I suspect that this is happening because of a bug fix that we made in the ImageMagick library. The image only contains two colors and that is why we are now only saving those two colors to the color map.
